packages/salesforcedx-vscode-apex-debugger/test/integration/adapter.test.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { expect } from 'chai';
+import { DebugClient } from 'vscode-debugadapter-testsupport';
```

Comment:
This utility behaves like a debugger client by launching the debug adapter and communicating with it. Nothing in the adapter can be mocked. Since there is no valid SFDX workspace and a scratch org to work with, there are only tests for basic initialize & attach requests.